src/main/java/vv/pms/ui/ProjectUI.java

```diff
@@ -125,7 +125,7 @@ public String projectDetails(@PathVariable Long id, Model model) {
                     profEntity.getId(), profEntity.getName(), profEntity.getEmail());
 
             // Get Students
-            Set<Long> studentIds = allocation.getAssignedStudentIds();
+            Set<Long> studentIds = (Set<Long>) allocation.getAssignedStudentIds();
```

Comment:
I see the getAssignedStudentIds was changed to a List<long> and added a check to maintain the only 1 student in the list requirement. Now that getAssignedStudentIds returns a List<long> this type cast here throws an exception:
```
java.lang.ClassCastException: class org.hibernate.collection.spi.PersistentBag cannot be cast to class java.util.Set 
at vv.pms.ui.ProjectUI.projectDetails(ProjectUI.java:128)
```

You can fix this problem by replaced Line 128 and Line 129 with this:
```
        List<Long> studentIds = allocation.getAssignedStudentIds();
        List<Student> studentEntities = (studentIds == null || studentIds.isEmpty()) ? Collections.emptyList() :
                List.copyOf(studentService.findByIds(Set.copyOf(studentIds)).values());
```